src/unitxt/task.py

```diff
@@ -38,20 +39,49 @@ class Task(InstanceOperator):
         "metrics" -- to contain the value of Arg 'metrics'
     """
 
-    input_fields: Union[Dict[str, str], List[str]]
-    reference_fields: Union[Dict[str, str], List[str]]
+    input_fields: Optional[Union[Dict[str, str], List[str]]] = None
+    reference_fields: Optional[Union[Dict[str, str], List[str]]] = None
+    inputs: Union[Dict[str, str], List[str]] = DeprecatedField(
+        default=None,
+        metadata={
+            "deprecation_msg": "The 'inputs' field is deprecated. Please use 'input_fields' instead."
+        },
+    )
+    outputs: Union[Dict[str, str], List[str]] = DeprecatedField(
+        default=None,
+        metadata={
+            "deprecation_msg": "The 'outputs' field is deprecated. Please use 'reference_fields' instead."
+        },
+    )
     metrics: List[str]
     prediction_type: Optional[str] = None
     augmentable_inputs: List[str] = []
     defaults: Optional[Dict[str, Any]] = None
 
+    def prepare(self):
+        super().prepare()
+        if (self.input_fields is not None and self.reference_fields is not None) and (
```

Comment:
I suggest separating into two checks, because people can mix only one pair of fields.

if (self.input_fields is not None and self.inputs is not None):
  aise ValueError(
                "Conflicting attributes: 'input_fields'  cannot be set simultaneously with 'inputs'. Use only 'input_fields'"
            )